source/promise-value.d.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Returns the type that is wrapped inside a `Promise` type.
+ * If the type is not a `Promise`, the type itself is returned.
+ *
+ * @example
+ * import { asyncFunction } from 'api';
+ * import { PromiseValue } from 'type-fest';
+ *
+ * let data: PromiseValue<ReturnType<typeof asyncFunction>>;
+ *
+ * async function setValue () {
+ *   data = await asyncFunction();
+ * }
+ *
+ * setValue();
+ *
+ * // Here's an example of using this with non-Promise types.
+ * function getNumber () {
+ *  return 2;
+ * }
```

Comment:
I’m not sure if other types follow this, but, could you lint this piece of code with XO? The style doesn’t pass it